tests/baselines/reference/jsDeclarationsJSDocRedirectedLookups.js

```diff
@@ -0,0 +1,82 @@
+//// [index.js]
+// these are recognized as TS concepts by the checker
+/** @type {String} */const a = "";
+/** @type {Number} */const b = 0;
+/** @type {Boolean} */const c = true;
+/** @type {Void} */const d = undefined;
+/** @type {Undefined} */const e = undefined;
+/** @type {Null} */const f = null;
+
+/** @type {Function} */const g = () => void 0;
+/** @type {function} */const h = () => void 0;
+/** @type {array} */const i = [];
+/** @type {promise} */const j = Promise.resolve(0);
+/** @type {Object<string, string>} */const k = {x: "x"};
+
+
+// these are not recognized as anything and should just be lookup failures
+// ignore the errors to try to ensure they're emitted as `any` in declaration emit
+// @ts-ignore
+/** @type {class} */const l = true;
+// @ts-ignore
+/** @type {bool} */const m = true;
+// @ts-ignore
+/** @type {int} */const n = true;
+// @ts-ignore
+/** @type {float} */const o = true;
+// @ts-ignore
+/** @type {integer} */const p = true;
+
+// or, in the case of `event` likely erroneously refers to the type of the global Event object
+/** @type {event} */const q = undefined;
+
+//// [index.js]
+"use strict";
+// these are recognized as TS concepts by the checker
+/** @type {String} */ const a = "";
+/** @type {Number} */ const b = 0;
+/** @type {Boolean} */ const c = true;
+/** @type {Void} */ const d = undefined;
+/** @type {Undefined} */ const e = undefined;
+/** @type {Null} */ const f = null;
+/** @type {Function} */ const g = () => void 0;
+/** @type {function} */ const h = () => void 0;
+/** @type {array} */ const i = [];
+/** @type {promise} */ const j = Promise.resolve(0);
+/** @type {Object<string, string>} */ const k = { x: "x" };
+// these are not recognized as anything and should just be lookup failures
+// ignore the errors to try to ensure they're emitted as `any` in declaration emit
+// @ts-ignore
+/** @type {class} */ const l = true;
+// @ts-ignore
+/** @type {bool} */ const m = true;
+// @ts-ignore
+/** @type {int} */ const n = true;
+// @ts-ignore
+/** @type {float} */ const o = true;
+// @ts-ignore
+/** @type {integer} */ const p = true;
+// or, in the case of `event` likely erroneously refers to the type of the global Event object
+/** @type {event} */ const q = undefined;
+
+
+//// [index.d.ts]
+/** @type {String} */ declare const a: string;
+/** @type {Number} */ declare const b: number;
+/** @type {Boolean} */ declare const c: boolean;
+/** @type {Void} */ declare const d: void;
+/** @type {Undefined} */ declare const e: undefined;
+/** @type {Null} */ declare const f: null;
+/** @type {Function} */ declare const g: Function;
+/** @type {function} */ declare const h: Function;
+/** @type {array} */ declare const i: any[];
+/** @type {promise} */ declare const j: Promise<any>;
+/** @type {Object<string, string>} */ declare const k: {

```

Comment:
What about `Object` => `object`?

Reply:
That's actually not a thing we interpret - `Object` with type arguments is interpreted as an index signature, without isn't recognized special (so it looks up the global `Object` constructor type, which is the apparent type of... pretty much everything except `null` and `undefined`) _except_ when `noImplicitAny` is false, where it is `any`.

`getIntendedTypeFromJSDocTypeReference` in the checker has the special jsdoc types that we lookup in it.

Reply:
Hmm, but the handbook says

> Don’t ever use the types Number, String, Boolean, Symbol, or Object These types refer to non-primitive boxed objects that are almost never used appropriately in JavaScript code.

> Instead of Object, use the non-primitive object type (added in TypeScript 2.2).

Reply:
The point is to test out stuff people write in JSDoc, because people write terrible stuff in JSDoc because it was terribly specified.